// angular2-image-upload/lib/image-upload.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var image_upload_component_1 = require("./image-upload/image-upload.component");
var file_drop_directive_1 = require("./file-drop.directive");
var common_1 = require("@angular/common");
var image_service_1 = require("./image.service");
var ImageUploadModule = (function () {
    function ImageUploadModule() {
    }
    ImageUploadModule.forRoot = function () {
        return {
            ngModule: ImageUploadModule,
            providers: [image_service_1.ImageService]
        };
    };
    ImageUploadModule = __decorate([
        core_1.NgModule({
            imports: [common_1.CommonModule],
            declarations: [
                image_upload_component_1.ImageUploadComponent,
                file_drop_directive_1.FileDropDirective
            ],
            exports: [image_upload_component_1.ImageUploadComponent]
        }), 
        __metadata('design:paramtypes', [])
    ], ImageUploadModule);
    return ImageUploadModule;
}());
exports.ImageUploadModule = ImageUploadModule;
